From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "akunkulagunt@" <akunkulagunt@>
Date: Wed, 8 Apr 2026 15:05:16 +0000
Subject: [PATCH 218/254] NVIDIA: SAUCE: ASoC: tegra: Add S8 PCM format support
 for MVC

Add handling for SNDRV_PCM_FORMAT_S8 in the MVC audio CIF
configuration. This allows 8-bit PCM streams to be processed
by setting the appropriate TEGRA_ACIF_BITS_8 format in the
CIF configuration.

Bug 5925800

Change-Id: I1aefdf75cf300a0f29143299d025cf373a32d0cd
Signed-off-by: Anupama Kunkulagunta <akunkulagunt@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3590354
(cherry picked from commit 3cdb2cb1f93dc41877ce51d4b43f43b7136823b7)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3596614
Reviewed-by: Mohan kumar <mkumard@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
APS-Inflight: svcenomis <svcenomis@nvidia.com>

diff --git a/sound/soc/tegra/tegra210_mvc.c b/sound/soc/tegra/tegra210_mvc.c
index ef8753ec48f9e..8dacc19d93b06 100644
--- a/sound/soc/tegra/tegra210_mvc.c
+++ b/sound/soc/tegra/tegra210_mvc.c
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0-only
-// SPDX-FileCopyrightText: Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES.
+// SPDX-FileCopyrightText: Copyright (c) 2021-2026 NVIDIA CORPORATION & AFFILIATES.
 // All rights reserved.
 //
 // tegra210_mvc.c - Tegra210 MVC driver
@@ -438,6 +438,9 @@ static int tegra210_mvc_set_audio_cif(struct tegra210_mvc *mvc,
 	channels = params_channels(params);
 
 	switch (params_format(params)) {
+	case SNDRV_PCM_FORMAT_S8:
+		audio_bits = TEGRA_ACIF_BITS_8;
+		break;
 	case SNDRV_PCM_FORMAT_S16_LE:
 		audio_bits = TEGRA_ACIF_BITS_16;
 		break;
-- 
2.43.0

